xdsl/dialects/arm/ops.py

```diff
@@ -0,0 +1,38 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+from xdsl.ir import Operation
+from xdsl.irdl import IRDLOperation, irdl_op_definition, result_def
+
+from .register import IntRegisterType
+
+R1InvT = TypeVar("R1InvT", bound=IntRegisterType)
```

Comment:
Are there plans to have more register types? This seems like a premature generalisation at this point

Reply:
Good point, we probably don't need it right now, let's add it later.